Adding Chocolatey state module

rdutch opened the pull request:
Chocolatey state module for installing and uninstalling chocolatey Windows packages.
The state will check if a package is already installed and if not install or visa versa for uninstalling.

Commit:
Adding Chocolatey state module

jfindlay commented:
@rdutch, there are some [lint errors](https://jenkins.saltstack.com/job/salt-pr-lint-n/7458/violations/file/salt/states/chocolatey.py/).

rdutch committed:
Update chocolatey.py

Fixed Pylint errors

rdutch commented:
Fixed pylint errors

cachedout commented on salt/states/chocolatey.py:
```diff
@@ -0,0 +1,162 @@
+        ret['comment'] = 'The Package {0} is already installed'.format(name)
+        return ret
+
+    if __opts__['test']:
```
We need to inform that user in the comments that this has been run in test mode. Otherwise, there is no indication to the user that the requested action has not taken place.

cachedout commented on salt/states/chocolatey.py:
```diff
@@ -0,0 +1,162 @@
+                                                             override_args,
+                                                             force_x86)}
+
+    if 'Running chocolatey failed' not in ret['changes']:
```
Are you certain this works? It looks like the exec module raises the error instead of returning it.

cachedout commented on salt/states/chocolatey.py:
```diff
@@ -0,0 +1,162 @@
+        return ret
+
+    if __opts__['test']:
+        ret['result'] = None
```
Same thing here. We need to inform the user what *would* have happened instead of just doing a no-op.

rallytime commented on salt/states/chocolatey.py:
```diff
@@ -0,0 +1,162 @@
+
+'''
+Manage Chocolatey package installs
+'''
```
Can you add a `.. versaionadded:: Boron` line this comment block?

cachedout commented:
@rdutch I left a few comments on this PR. Please have a look and let me know what you think. Then we'll get this merged. Thanks!

rdutch committed:
Fixed some issues

Fixed Pylint errors. Added a comment in the test mode.
In order to test the state, I had to add the develop chocolatey.py module to the \srv\salt\_modules folder on the master and do a sync_all.

cachedout commented:
Go Go Jenkins!

cachedout commented:
I linted this by hand and it is fine. Merging.
